tests/test_edge_cases.py

```diff
@@ -187,3 +187,29 @@ def test_complex_cell_parsing(lines):
         montepy.input_parser.block_type.BlockType.CELL,
     )
     montepy.Cell(input)
+
+def test_the_dissapearing_parens():
+    in_file = os.path.join("tests", "inputs", "test_paren_groups.imcnp")
+    problem = montepy.read_input(in_file)
+    parens_count = 0
+    with open(in_file, "r") as fh:
+        for line in fh:
+            for char in line:
+                if char in {"(", ")"}:
+                    parens_count += 1
+    try:
+        out_file = "test_parens"
+        problem.write_to_file(out_file)
+        new_parens_count = 0
+        with open(out_file, "r") as fh:
+            for line in fh:
+                print(line.rstrip())
+                for char in line:
+                    if char in {"(", ")"}:
+                        new_parens_count += 1
+        assert new_parens_count == parens_count
+    finally:
+        try:
+            os.remove(out_file)
+        except FileNotFoundError:
+            pass
```

Comment:
I'd like thoughts on how to fixture this try finally block. I write it a ton.

Reply:
[I have a better idea.](https://github.com/idaholab/MontePy/issues/492)

Reply:
Yeah before writing this I was pondering if `IO.StringIO` would work.

Reply:
It will 🤠